backend/attachRendererFiber.js

```diff
@@ -0,0 +1,244 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ */
+'use strict';
+
+import type {Hook, ReactRenderer, Helpers} from './types';
+var getDataFiber = require('./getDataFiber');
+
+function attachFiberRenderer(hook: Hook, rid: string, renderer: ReactRenderer): Helpers {
+  function getNativeFromReactElement() {
+    // TODO
+    // This will likely require using Fibers
+    // as "opaque" IDs instead of debugIDs,
+    // and exposing the host component tree again.
+    return null;
+  }
+
+  function getReactElementFromNative() {
+    // TODO
+    // This will likely require using Fibers
+    // as "opaque" IDs instead of debugIDs,
+    // and exposing the host component tree again.
+    return null;
+  }
+
+  function walkTree() {
+    // TODO
+    // So far we didn't need this in Fiber
+    // since we always walk the tree anyway.
+  }
+
+  function enqueueMount(fiber, events) {
+    events.push({
+      // TODO: we might need to pass Fiber instead to implement getNativeFromReactElement().
+      // TODO: the naming is confusing. `element` is *not* a React element. It is an opaque ID.
+      element: fiber._debugID,
+      data: getDataFiber(fiber),
+      renderer: rid,
+      _event: 'mount',
+    });
+    const isRoot = fiber.tag === 3;
+    if (isRoot) {
+      events.push({
+        element: fiber._debugID,
+        renderer: rid,
+        _event: 'root',
+      });
+    }
+  }
+
+  function enqueueUpdate(fiber, events) {
```

Comment:
I'd still need to compare children though, don't I? So that would be reference equality for everything except child ID array (which I'd compare shallowly).